src/core/dbus-execute.c

```diff
@@ -3563,6 +3594,42 @@ int bus_exec_context_set_transient_property(
 
                 return 1;
 
+        } else if (STR_IN_SET(name, "StateDirectoryQuota", "CacheDirectoryQuota", "LogsDirectoryQuota")) {
+                uint64_t quota_absolute = UINT64_MAX;
+                uint32_t quota_scale = UINT32_MAX;
+                int quota_enforce;
+
+                r = sd_bus_message_read(message, "(tub)", &quota_absolute, &quota_scale, &quota_enforce);
+                if (r < 0)
+                        return r;
+
+                if (!UNIT_WRITE_FLAGS_NOOP(flags)) {
+                        if (!quota_enforce)
+                                unit_write_settingf(u, flags, name, "%s=", name);
```

Comment:
you should set c->directories[dt].exec_quota.quota_enforce here too, i.e. actually disable things.